File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/OrderConnection.java
package com.example.grts;

import viaduct.engine.api.EngineObjectData;
import viaduct.engine.api.RootFieldReference;
import viaduct.java.api.context.ExecutionContext;
import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.internal.ConnectionBuilder;
import viaduct.java.api.internal.InternalContext;
import viaduct.java.api.internal.ObjectBase;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.RootObjectField;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.Arguments;
import viaduct.java.api.types.ConnectionArguments;
import viaduct.java.api.types.OffsetLimit;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.List;
import java.util.Map;
import java.util.function.Function;

@SuppressWarnings("MissingOverride")
public class OrderConnection extends ObjectBase implements viaduct.java.api.types.Connection<OrderEdge, Order> {

    public static final Type<OrderConnection> Reflection = Type.ofClass(OrderConnection.class);

    public static final class Fields implements TypeFields<OrderConnection> {
        private Fields() {}

        public static final Field<OrderConnection> __typename =
                Field.of("__typename", Reflection);
                public static final CompositeField<OrderConnection, OrderEdge> edges =
                                CompositeField.of("edges", Reflection, OrderEdge.Reflection);

                public static final CompositeField<OrderConnection, PageInfo> pageInfo =
                                CompositeField.of("pageInfo", Reflection, PageInfo.Reflection);

                public static final Field<OrderConnection> totalCount =
                                Field.of("totalCount", Reflection);

    }

    public OrderConnection(InternalContext context, EngineObjectData.Sync data) {
        super(context, data);
    }

    @SuppressWarnings("UnusedMethod")
    private OrderConnection(InternalContext context, Map<String, Object> data) {
        super(context, data, "OrderConnection");
    }

    private OrderConnection(InternalContext context, ObjectBase base, Map<String, Object> data) {
        super(context, base, data, "OrderConnection");
    }

    public OrderConnection(InternalContext context, RootFieldReference rootFieldReference) {
        super(context, rootFieldReference);
    }

        public List<OrderEdge> getEdgesOrThrow(String alias) {
            return fetchObjectList("edges", alias, OrderEdge.class, OrderEdge::new);
        }

        public List<OrderEdge> getEdgesOrThrow() {
            return fetchObjectList("edges", null, OrderEdge.class, OrderEdge::new);
        }

        public List<OrderEdge> getEdges(String alias) {
            return nullOnDataFailure(() -> fetchObjectList("edges", alias, OrderEdge.class, OrderEdge::new));
        }

        public List<OrderEdge> getEdges() {
            return nullOnDataFailure(() -> fetchObjectList("edges", null, OrderEdge.class, OrderEdge::new));
        }

        public PageInfo getPageInfoOrThrow(String alias) {
            return fetchObject("pageInfo", alias, PageInfo.class, PageInfo::new);
        }

        public PageInfo getPageInfoOrThrow() {
            return fetchObject("pageInfo", null, PageInfo.class, PageInfo::new);
        }

        public PageInfo getPageInfo(String alias) {
            return nullOnDataFailure(() -> fetchObject("pageInfo", alias, PageInfo.class, PageInfo::new));
        }

        public PageInfo getPageInfo() {
            return nullOnDataFailure(() -> fetchObject("pageInfo", null, PageInfo.class, PageInfo::new));
        }

        public Integer getTotalCountOrThrow(String alias) {
            return fetchScalar("totalCount", alias);
        }

        public Integer getTotalCountOrThrow() {
            return fetchScalar("totalCount", null);
        }

        public Integer getTotalCount(String alias) {
            return nullOnDataFailure(() -> fetchScalar("totalCount", alias));
        }

        public Integer getTotalCount() {
            return nullOnDataFailure(() -> fetchScalar("totalCount", null));
        }


    public Builder toBuilder() {
        return new Builder(__context(), toBuilderBase());
    }

    public static Builder builder(ExecutionContext context) {
        return new Builder(context);
    }

    /**
     * Pagination-aware builder. Extends {@link ConnectionBuilder} with {@code fromEdges},
     * {@code fromSlice}, and {@code fromList}; the base builds the OrderEdge,
     * PageInfo, and OrderConnection GRTs from the field's context and these Class handles.
     * The generated per-field setters populate additional connection fields (e.g.
     * {@code totalCount}) alongside the pagination-produced {@code edges}/{@code pageInfo};
     * a pagination method and any setters can be combined in any order before {@code build()}.
     */
    public static class Builder extends ConnectionBuilder<OrderConnection, OrderEdge, Order> {
        private final ObjectBase __base;

        private Builder(ExecutionContext context) {
            this(InternalContext.from(context), null);
        }

        private Builder(InternalContext context, ObjectBase base) {
            super(OrderConnection.class, OrderEdge.class, context);
            this.__base = base;
        }

        @Override
        public OrderConnection build() {
            return new OrderConnection(internalContext, __base, buildData());
        }

        @Override
        public Builder fromEdges(List<OrderEdge> edges) {
            super.fromEdges(edges);
            return this;
        }

        @Override
        public Builder fromEdges(
                List<OrderEdge> edges,
                boolean hasNextPage,
                boolean hasPreviousPage) {
            super.fromEdges(edges, hasNextPage, hasPreviousPage);
            return this;
        }

        @Override
        public <I> Builder fromSlice(
                List<I> items,
                boolean hasNextPage,
                Function<I, Order> buildNode) {
            super.fromSlice(items, hasNextPage, buildNode);
            return this;
        }

        @Override
        public <I> Builder fromSlice(
                List<I> items,
                OffsetLimit offsetLimit,
                boolean hasNextPage,
                Function<I, Order> buildNode) {
            super.fromSlice(items, offsetLimit, hasNextPage, buildNode);
            return this;
        }

        @Override
        public <I> Builder fromList(
                List<I> items,
                Function<I, Order> buildNode) {
            super.fromList(items, buildNode);
            return this;
        }

                public Builder edges(List<OrderEdge> edges) {
                    putField(
                                        "edges",
                                        edges,
                                        OrderEdge.class);
        return this;
                }

                public Builder pageInfo(PageInfo pageInfo) {
                    putField(
                                        "pageInfo",
                                        pageInfo,
                                        PageInfo.class);
        return this;
                }

                public Builder totalCount(Integer totalCount) {
                    putField(
                                        "totalCount",
                                        totalCount,
                                        null);
        return this;
                }

    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/OrderConnection.kt
@file:Suppress("warnings")

package viaduct.api.grts

import viaduct.apiannotations.ExperimentalApi
import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.ConnectionBuilder
import viaduct.api.internal.InternalContext
import viaduct.api.internal.ObjectBase
import viaduct.engine.api.EngineObject
import viaduct.engine.api.EngineObjectData

@OptIn(InternalApi::class, ExperimentalApi::class)
class OrderConnection(context: InternalContext, engineObject: EngineObject)
    : ObjectBase(context, engineObject), viaduct.api.types.Object,viaduct.api.types.Connection<viaduct.api.grts.OrderEdge, viaduct.api.grts.Order>
{
     fun getEdgesOrThrow(alias: String?): kotlin.collections.List<viaduct.api.grts.OrderEdge> = TODO()
     fun getEdgesOrThrow(): kotlin.collections.List<viaduct.api.grts.OrderEdge> = TODO()
     fun getEdges(alias: String?): kotlin.collections.List<viaduct.api.grts.OrderEdge>? = TODO()
     fun getEdges(): kotlin.collections.List<viaduct.api.grts.OrderEdge>? = TODO()

     fun getPageInfoOrThrow(alias: String?): viaduct.api.grts.PageInfo? = TODO()
     fun getPageInfoOrThrow(): viaduct.api.grts.PageInfo? = TODO()
     fun getPageInfo(alias: String?): viaduct.api.grts.PageInfo? = TODO()
     fun getPageInfo(): viaduct.api.grts.PageInfo? = TODO()


    fun toBuilder(): Builder =
        Builder(__context, __engineObject.type, toBuilderEOD())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): OrderConnection =
            Builder(context).apply(block).build()
    }

    class Builder : ConnectionBuilder<OrderConnection, viaduct.api.grts.OrderEdge, viaduct.api.grts.Order> {
        constructor(context: ExecutionContext)
            : super(
                context,
                TODO() as graphql.schema.GraphQLObjectType,
                null,
                viaduct.api.grts.OrderEdge.Reflection
            )

        internal constructor(
            context: InternalContext,
            type: graphql.schema.GraphQLObjectType,
            baseEngineObjectData: EngineObjectData.Sync
        ) : super(
                context as ExecutionContext,
                type,
                baseEngineObjectData,
                viaduct.api.grts.OrderEdge.Reflection
            )

                  fun edges(value: kotlin.collections.List<viaduct.api.grts.OrderEdge>): Builder = TODO()

                  fun pageInfo(value: viaduct.api.grts.PageInfo?): Builder = TODO()


        final override fun build(): OrderConnection {
            @Suppress("UNUSED_EXPRESSION")
            ObjectBase.Builder::class
            return TODO()
        }
    }

    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.OrderConnection> {
        override final val name = "OrderConnection"
        override final val kcls = viaduct.api.grts.OrderConnection::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.OrderConnection> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.OrderConnection> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.OrderConnection.Reflection)

            final val edges: viaduct.api.reflect.CompositeField<viaduct.api.grts.OrderConnection, viaduct.api.grts.OrderEdge> =
                viaduct.api.internal.CompositeFieldImpl("edges", viaduct.api.grts.OrderConnection.Reflection, viaduct.api.grts.OrderEdge.Reflection)

            final val pageInfo: viaduct.api.reflect.CompositeField<viaduct.api.grts.OrderConnection, viaduct.api.grts.PageInfo> =
                viaduct.api.internal.CompositeFieldImpl("pageInfo", viaduct.api.grts.OrderConnection.Reflection, viaduct.api.grts.PageInfo.Reflection)

    }

}